src/Clerk/Provider.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace SocialiteProviders\Clerk;
+
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    public const IDENTIFIER = 'CLERK';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = ['profile','email'];
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopeSeparator = ' ';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getBaseUrl()
+    {
+        return $this->getConfig('base_url');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function additionalConfigKeys()
+    {
+        return ['base_url'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase("{$this->getBaseUrl()}/oauth/authorize", $state);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenUrl()
+    {
+        return "{$this->getBaseUrl()}/oauth/token";
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getUserByToken($token)
+    {
+        $response = $this->getHttpClient()->get("{$this->getBaseUrl()}/oauth/userinfo", [
+            'headers' => [
+                'Authorization' => 'Bearer '.$token,
+            ],
+        ]);
+        return json_decode($response->getBody(), true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function mapUserToObject(array $user)
+    {
+        return (new User())->setRaw($user)->map([
+            'id'       => $user['user_id'],
+            'nickname' => $user['username'],
+            'name'     => $user['name'],
+            'email'    => $user['email'],
+            'avatar'   => $user['picture'],
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenFields($code)
```

Comment:
Is this needed or the same as the parent method?

Reply:
@atymic it's been addressed